src/components/SelectComponents/Combobox/__tests__/Combobox.spec.tsx

```diff
@@ -17,178 +19,209 @@ describe('Combobox', () => {
+                const newProps = {
+                    suggestions,
+                    suggestionToString,
+                    inputPlaceholder,
+                    noSuggestionsPlaceholder,
+                    onSelectionAdd: mockOnSelectionAdd,
+                    onInputValueChange: mockOnInputValueChange,
+                    onBlur: mockOnBlur,
+                    selectedSuggestion: SUGGESTIONS[1],
+                };
+                rerenderView(newProps);
+                await setFocus();
+
+                expect(inputField.getAttribute('placeholder')).toEqual(SUGGESTIONS[1].name);
             });
         });
     });
     describe('focusing and blurring the search field', () => {
-        it('should set focus when wrapper element is clicked', () => {
-            expect(getInputNode()).not.toBe(document.activeElement);
+        it('should set focus when wrapper element is clicked', async () => {
+            expect(getInputNode(viewContainer)).not.toBe(document.activeElement);
+
+            await setFocus();
 
-            setFocusOnInput();
-            expect(getInputNode()).toBe(document.activeElement);
+            expect(screen.getByRole('textbox')).toBe(document.activeElement);
         });
-        it('should not stay focused when suggestion is selected', () => {
-            expect(getInputNode()).not.toBe(document.activeElement);
+        it('should not stay focused when suggestion is selected', async () => {
+            expect(getInputNode(viewContainer)).not.toBe(document.activeElement);
 
-            setFocusOnInput();
-            expect(getInputNode()).toBe(document.activeElement);
+            await setFocus();
+            expect(getInputNode(viewContainer)).toBe(document.activeElement);
 
-            wrapper.find('li').at(0).childAt(0).simulate('click');
+            await userEvent.click(screen.queryAllByRole('presentation')[0]);
 
-            expect(getInputNode()).not.toBe(document.activeElement);
-            expect(wrapper.find('li')).toHaveLength(0);
+            expect(getInputNode(viewContainer)).not.toBe(document.activeElement);
+            expect(screen.queryAllByRole('presentation')).toHaveLength(0);
         });
-        it('should blur on pressing Escape button', () => {
-            setFocusOnInput();
-            expect(getInputNode()).toBe(document.activeElement);
+        it('should blur on pressing Escape button', async () => {
+            await setFocus();
+            expect(getInputNode(viewContainer)).toBe(document.activeElement);
 
-            wrapper.find('input').simulate('keyDown', { key: 'Escape' });
-            wrapper.update();
+            await userEvent.keyboard('[Escape]');
 
-            expect(getInputNode()).not.toBe(document.activeElement);
+            expect(getInputNode(viewContainer)).not.toBe(document.activeElement);
             // TODO: fixMe - the callback is not triggered, even though it does work in the UI
```

Comment:
please delete this TODO if it is working now as expected. Thanks

Reply:
:+1: